File: generate_screenshots.py
from playwright.sync_api import sync_playwright
import os

sizes = {
    "6.5-inch": {"width": 1284, "height": 2778},
    "5.5-inch": {"width": 1242, "height": 2208}
}

base_url = "file://" + os.path.abspath("appstore_gen.html")

if not os.path.exists("screenshots"):
    os.makedirs("screenshots")

with sync_playwright() as p:
    browser = p.chromium.launch()
    for size_name, dim in sizes.items():
        page = browser.new_page(viewport={"width": dim["width"], "height": dim["height"]})
        for i in range(4):
            page.goto(f"{base_url}?i={i}")
            # wait for fonts and images to load
            page.wait_for_load_state("networkidle")
            page.wait_for_timeout(500) # give it a moment to render fonts
            
            output_path = f"screenshots/{size_name}_{i+1}.png"
            page.screenshot(path=output_path)
            print(f"Generated {output_path}")
    browser.close()

File: generate_screenshots.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const sizes = {
    "6.7-inch": { width: 1290, height: 2796 },
    "6.5-inch": { width: 1284, height: 2778 },
    "5.5-inch": { width: 1242, height: 2208 }
};

const baseUrl = "file://" + path.resolve(__dirname, "appstore_gen.html");

if (!fs.existsSync("screenshots")) {
    fs.mkdirSync("screenshots");
}

(async () => {
    const browser = await chromium.launch();
    
    for (const [sizeName, dim] of Object.entries(sizes)) {
        const page = await browser.newPage({ viewport: dim });
        for (let i = 0; i < 4; i++) {
            await page.goto(`${baseUrl}?i=${i}`);
            await page.waitForLoadState('networkidle');
            await page.waitForTimeout(1000); // Wait for fonts/images
            
            const outputPath = path.join("screenshots", `${sizeName}_${i+1}.png`);
            await page.screenshot({ path: outputPath });
            console.log(`Generated ${outputPath}`);
        }
    }
    await browser.close();
})();